### PR TITLE
Hotfix/peer 99

### PR DESCRIPTION
**Pull request by YuriyLu**

### CODE REVIEW

File: contracts/libraries/PostLib.sol [hunk 1]
```diff
@@ -331,15 +332,22 @@ library PostLib  {
             replyContainer.info.author,
             postContainer.info.communityId,
             UserLib.Action.EditItem,
-            UserLib.ActionRole.NONE,
+            isOfficialReply ? UserLib.ActionRole.CommunityModerator :
```
Comment (Freitag-fri) [inline contracts/libraries/PostLib.sol:334]: user without moderator rights can unmark official reply.
Will add unit test `giveCommunityModeratorPermission(user, ...) -> publication official reply -> revokeCommunityModeratorPermission(user, ...) ->  editReply (..., isOfficialReply false)`

File: contracts/libraries/PostLib.sol [hunk 2]
```diff
@@ -331,15 +332,22 @@ library PostLib  {
...
         require(userAddr == replyContainer.info.author, "You can not edit this Reply. It is not your.");
 
         if (replyContainer.info.ipfsDoc.hash != ipfsHash)
             replyContainer.info.ipfsDoc.hash = ipfsHash;
-        
+
+        if (isOfficialReply) {
+            postContainer.info.officialReply = replyId;
+        } else {
+            postContainer.info.officialReply = 0;
```
Comment (Freitag-fri) [inline contracts/libraries/PostLib.sol:347]: When common user edits common reply, he unmarks official status for another reply
Will add unit test

**Review by Freitag-fri — CHANGES_REQUESTED** [review]

File: contracts/libraries/PostLib.sol (hunk 2 above)
Comment (YuriyLu) [reply contracts/libraries/PostLib.sol:347]: Fixed

File: contracts/libraries/PostLib.sol (hunk 1 above)
Comment (YuriyLu) [reply contracts/libraries/PostLib.sol:334]: Answered in slack

Comment (sergeyi23) [reply contracts/libraries/PostLib.sol:334]: What is the resolution here? I think it is still an issue.

**Review by sergeyi23 — APPROVED** [review]

**Review by sergeyi23 — CHANGES_REQUESTED** [review]
This seems to be still a problem - https://github.com/peeranha/peeranha/pull/67/files#r902120524

**Comment by sergeyi23** [comment]
Event `StatusOfficialReplyChanged` is not used anymore. Should be removed if not needed.

File: contracts/libraries/PostLib.sol (hunk 1 above)
Comment (YuriyLu) [reply contracts/libraries/PostLib.sol:334]: In my opinion, after the moderation rights removing user shouldn't be able to post the answers with "Official answer" flag, the same with editing the existing replies.

Comment (sergeyi23) [reply contracts/libraries/PostLib.sol:334]: Per discussion offline, if a user is no longer a moderator and edits a question that was official we need to uncheck the official flag automatically. Logic is correct.

**Review by sergeyi23 — APPROVED** [review]

**Pull request merged**